frontend/src/Judge/reducers.js

```diff
@@ -2,6 +2,97 @@ import { combineReducers } from 'redux'
 
 import * as actions from './actions'
 
+// Example State:
+// {
+//   1: {id: '1', title...},
+//   2: {id: '2', title...}
+// }
+const submissions = (state = {}, action) => {
+  switch (action.type) {
+    case actions.FETCH_SUBMISSION:
+      if (!action.payload.id) {
+        return state
+      }
+
+      return {
+        ...state,
+        [action.payload.id]: {
+          ...state[action.payload.id],
+          ...action.payload
+        }
+      }
+    case actions.FETCH_SUBMISSIONS:
+      if (!action.payload.length) {
+        return state
+      }
+
+      const submissions = action.payload.reduce((accum, submission) => {
+        accum[submission.id] = submission
+        return accum
+      }, {})
+      return {
+        ...state,
+        ...submissions
+      }
+    default:
+      return state
+  }
+}
+
+// TODO: Produce 'order' based on shuffling the entry id's that are in this show
+// TODO: Calculate 'viewing' based on the first unvoted entry in 'order'
+const initialQueueState = {
+  order: [], // Array of entry id's
+  viewing: 0 // The index in 'order' we're currently viewing
+}
+
+const queue = (state = initialQueueState, action) => {
+  const { order, viewing } = state
+
+  switch (action.type) {
+    case actions.NEXT_IN_QUEUE:
+      return {
+        ...state,
+        // Continue incrementing until we've reached the end of the list.
+        // Then continue to return the index of the last element.
+        viewing: viewing < order.length ? viewing + 1 : viewing
```

Comment:
Can this extend beyond the array...? Seems like when `viewing = order.length-1` then `viewing` will be set to `order.length`, which would be an out-of-bounds?

Maybe just `viewing: Math.min(order.length-1, viewing+1)`

Reply:
Yes. I'll update it. `viewing + 1 < order.length` might be easier. `Math.min` will be -1 if `length` is 0.

Something, something, we should test our reducers...